jwst/pipeline/calwebb_spec2.py

```diff
@@ -268,6 +268,15 @@ def process_exposure_product(
 
         # apply the "nsclean" 1/f correction to NIRSpec images
         calibrated = self.nsclean(calibrated)
+        
+        # Apply nsclean to NIRSpec imprint and background members
+        for i, imprint_file in enumerate(members_by_type['imprint']):
+            imprint_nsclean = self.nsclean(imprint_file)
+            members_by_type['imprint'][i] = imprint_nsclean
+            
+        for i, bkg_file in enumerate(members_by_type['background']):
+            bkg_nsclean = self.nsclean(bkg_file)
+            members_by_type['background'][i] = bkg_nsclean
```

Comment:
Testing locally, it looks like this runs okay, and saving the intermediate files also works fine, but passing `--steps.nsclean.save_results` uses the science filename for all of the products, so the saved product is overwritten.

I'm investigating what we might need to do to override the output name for the different input files.

Reply:
I think setting the `output_file` for nsclean before calling it should work:

```suggestion
        # Apply nsclean to NIRSpec imprint and background members
        for i, imprint_file in enumerate(members_by_type['imprint']):
            self.nsclean.output_file = os.path.basename(imprint_file)
            imprint_nsclean = self.nsclean(imprint_file)
            members_by_type['imprint'][i] = imprint_nsclean
            
        for i, bkg_file in enumerate(members_by_type['background']):
            self.nsclean.output_file = os.path.basename(bkg_file)
            bkg_nsclean = self.nsclean(bkg_file)
            members_by_type['background'][i] = bkg_nsclean
```